File: app/src/main/java/com/alexjlockwood/example/delight/CheckableActivity.java
package com.alexjlockwood.example.delight;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class CheckableActivity extends AppCompatActivity {

  @BindView(R.id.radiobutton) ImageView radioButtonView;
  @BindView(R.id.checkbox) ImageView checkBoxView;
  @BindView(R.id.expandcollapse) ImageView expandCollapseView;
  private boolean isChecked;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_checkable);
    ButterKnife.bind(this);
  }

  @OnClick(R.id.rootview)
  void onClick() {
    isChecked = !isChecked;
    final int[] stateSet = {android.R.attr.state_checked * (isChecked ? 1 : -1)};
    radioButtonView.setImageState(stateSet, true);
    checkBoxView.setImageState(stateSet, true);
    expandCollapseView.setImageState(stateSet, true);
  }
}

File: app/src/main/java/com/alexjlockwood/example/delight/CheckableActivity.kt
package com.alexjlockwood.example.delight

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.ImageView
import butterknife.BindView
import butterknife.ButterKnife
import butterknife.OnClick

class CheckableActivity : AppCompatActivity() {
    @BindView(R.id.radiobutton)
    var radioButtonView: ImageView? = null

    @BindView(R.id.checkbox)
    var checkBoxView: ImageView? = null

    @BindView(R.id.expandcollapse)
    var expandCollapseView: ImageView? = null
    private var isChecked = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_checkable)
        ButterKnife.bind(this)
    }

    @OnClick(R.id.rootview)
    fun onClick() {
        isChecked = !isChecked
        val stateSet = intArrayOf(android.R.attr.state_checked * if (isChecked) 1 else -1)
        radioButtonView!!.setImageState(stateSet, true)
        checkBoxView!!.setImageState(stateSet, true)
        expandCollapseView!!.setImageState(stateSet, true)
    }
}